### PR TITLE
Added guidelines for contributors

### LINKED ISSUE #500: Create a CONTRIBUTING.md
- Describe format for PRs
- Describe expectations for individual commits in PRs

### PR DESCRIPTION
**Pull request by ciyer**
Fix #500

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- Detailed steps to reproduce the bug.
+
+Pull Requests
+=============
+
+If you do not already have a development environment set up, you will probably find the
+`developer documentation`_ helpful.
+
+* Before submitting a pull request, please make sure you agree to the license and have submitted a signed `contributor license agreement`_.
+* PRs should include a short, descriptive title. The titles will be used to compile changelogs for releases, so think about the title in that context.
```
Comment (jirikuncar) [inline CONTRIBUTING.rst]: > The titles will be used to compile changelogs for releases, so think about the title in that context.

Do we really want to use PR titles?

Comment (rokroskar) [reply CONTRIBUTING.rst]: We agreed IRL that at first we would do the changelogs by hand anyway from a combination of PR titles and commit messages. Still, pretending like the PR title will be used by a robot in a changelog might encourage better PR titles :)

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
+Contributing to Renku
+=====================
+
+Want to contribute to Renku? Thanks! There are many ways to help, and we very much
+appreciate your efforts.
+
+The sections below provide guidelines for various types of contributions.
+
+Bug Reports and Feature Requests
+===========
```
Comment (rokroskar) [inline CONTRIBUTING.rst]: The underline has to be the length of the text above otherwise Sphinx complains

Comment (ciyer) [reply CONTRIBUTING.rst]: Fixed.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (jirikuncar) [reply CONTRIBUTING.rst]: We should encourage ourselves to write nice commit message in case the commits are indexed in Renku itself one day ;)

**Pull request merged**